float64 bearing
float64 theta
